integration/extensions/test.ts

```diff
@@ -44,19 +44,19 @@ export function enumToJSON(object: Enum): string {
 }
 
 export interface Extendable {
-  field?: string;
-  _unknownFields?: { [key: number]: Uint8Array[] };
+  field?: string | undefined;
```

Comment:
@threema-lenny is this just a stylistic preference? I thought `field?: string` and `field?: string | undefined` were semantically equivalent.

Reply:
Unfortunately they aren't because there's a semantic difference between _presence with `undefined`_ and _no presence_ of a property. [The TS config reference for `exactOptionalPropertyTypes`](https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes) enforces that semantic difference and also has a short example for it.

Reply:
Huh, I knew about that difference wrt runtime behavior (observing lack-of-presence vs. present-and-undefined), but didn't know about `exactOptionalPropertyTypes`.

I've also just used `strict: true` and assumed I was getting the most strict checking, so am surprised to learn about this `strictest` setting that turns on even a few more things--but it explains why I'd never been tripped up by `foo?: string` vs `foo?: string | undefined` before.

Sounds good then, thanks!